### PR TITLE
feat: Add EP0.7 audited MSCA contracts testnet deployment logs and update scripts

### PR DESCRIPTION
**Pull request by ashutosh-ukey**
## Summary

Update deployment scripts and add deployment logs for audited MSCA EP0.7 contracts. 

### Checklist
- [ ] Did you add new tests and confirm all tests pass? (`yarn test`)
- [x] Did you update relevant docs? (docs are found in the `docs` folder)
- [ ] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [x] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples)? (e.g. `feat!: breaking change`)
- [x] Did you run lint (`yarn lint`) and fix any issues?
- [x] Did you run formatter (`yarn format:check`) and fix any issues (`yarn format:write`)?

## Testing

All the script changes were tested locally in the process of deploying the new contracts. Any edits made during the PR review process were tested by simulating the script and verifying that there were no errors. I tested the `cast send` commands on a local networks and verified that contracts were deployed at the same addresses as those deployed using the `forge script` command.

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -1,17 +1,26 @@
...
 ETHERSCAN_API_KEY=
+
+########### Contract addresses ###########
+# Set to "0x0000000000000000000000000000000000000000" if unknown or deploying new version
+PLUGIN_MANAGER_ADDRESS=0x0000000000000000000000000000000000000000
+UPGRADABLE_MSCA_FACTORY_ADDRESS=0x0000000000000000000000000000000000000000
+SINGLE_OWNER_PLUGIN_ADDRESS=0x0000000000000000000000000000000000000000
+COLD_STORAGE_ADDRESS_BOOK_PLUGIN_ADDRESS=0x0000000000000000000000000000000000000000
+WEIGHTED_MULTISIG_PLUGIN_ADDRESS=0x0000000000000000000000000000000000000000
+DEFAULT_TOKEN_CALLBACK_PLUGIN_ADDRESS=0x0000000000000000000000000000000000000000
```
Comment (huaweigu) [inline .env.example:26]: new line?

File: README.md [hunk 2]
```diff
@@ -1,45 +1,53 @@
-# buidl-wallet-contracts
+# smart-wallet-contracts
```
Comment (huaweigu) [inline README.md:1]: revert this?

File: .env.example [hunk 3]
```diff
@@ -1,17 +1,26 @@
...
 DEPLOYER_PRIVATE_KEY=0x8...6
 DEPLOYER_ADDRESS=0xB...E
 
-
 ########### API keys ###########
 INFURA_API_KEY=
 ALCHEMY_API_KEY=
 ETHERSCAN_API_KEY=
+
+########### Contract addresses ###########
```
Comment (huaweigu) [inline .env.example:19]: Consider moving these to a Solidity file?
We will have more addresses as the product iterates

File: script/011.1_SetUpgradableMSCAFactoryPlugins.s.sol [hunk 4]
```diff
@@ -0,0 +1,63 @@
+/*
```
Comment (dlim-circle) [inline script/011.1_SetUpgradableMSCAFactoryPlugins.s.sol:1]: What's the difference between this and script 009? 
Is there a meaning on the 1_ in front of the script name ?

File: script/cmd/ColdStorageAddressBookPlugin [hunk 5]
```diff
@@ -0,0 +1 @@
...e01b0319611ac182611968565b1663a22cb46560e01b81148015611b43575b15611ae9575060448151106119e1576024015190565b632142170760e11b81148015611b33575b15611b10575060648151106119e1576044015190565b635c46a7ef60e11b14611b235750600090565b60a48151106119e1576044015190565b506323b872dd60e01b8114611afa565b5063095ea7b360e01b8114611ad3565b9061ffff1916908115611bdb57604080516001600160a01b039092168252637c9c64bb60e11b602083015260009082015260608101828152608082208054909290611bd257600160809252209182549283158015611bc8575b15611bbc57600193505555600190565b90600217905555600190565b5060018416611bac565b50505050600090565b5050600090565b906040519160018060a01b03168252637c9c64bb60e11b602083015260008060408401526000906060840160019460018252608090608081205492604051948860808701995b611c45575b5050505050506001820160051b016080016040528252565b9091929394968188161580611c9b575b15611c955761fffe198816908201600581901b880187018290529760021615611c8c5783528085852054915b959493929195611c28565b50808291611c81565b96611c2d565b50871515611c55565b805160005b818110611cb557505090565b80611cc260019285611276565b5160601c611cd08286611276565b5201611ca956fea26469706673582212209a2412a8aeab08683f5199b932bfa3c1fa159ee5c0a0a0e0d878a0bd2f99505664736f6c63430008180033
```
Comment (dlim-circle) [inline script/cmd/ColdStorageAddressBookPlugin:1]: I see all the runs on all chains have json output - does it mean you did forge deploy for all contracts ? Were we able to test these cast commands on any chain?

File: script/011.1_SetUpgradableMSCAFactoryPlugins.s.sol (hunk 4 above)
Comment (huaweigu) [reply script/011.1_SetUpgradableMSCAFactoryPlugins.s.sol:1]: It borrows the idea from flyway :)

Comment (ashutosh-ukey) [reply script/011.1_SetUpgradableMSCAFactoryPlugins.s.sol:1]: The difference between these files is mentioned as a code comment at the top of this file:

```
// Only needed for MSCA v0.7 factories. This script is the same as 009_SetUpgradableMSCAFactoryPlugins.s.sol, but with
// the
// default token callback plugin included).
```

File: script/cmd/ColdStorageAddressBookPlugin (hunk 5 above)
Comment (ashutosh-ukey) [reply script/cmd/ColdStorageAddressBookPlugin:1]: Yes, I used forge deploy for all of the contracts across all of the chains, so I have not tested these commands. But, I did construct them using the instructions specified in the README. Should I test them on a local network to confirm I get the same address?

Comment (dlim-circle) [reply script/cmd/ColdStorageAddressBookPlugin:1]: Yeah I usually forge deploy with 1 chain and use this for rest of the chains - as testing for chain expansion later

Comment (ashutosh-ukey) [reply script/cmd/ColdStorageAddressBookPlugin:1]: Just curious, does cast send also verify the contracts like forge deploy?

Comment (dlim-circle) [reply script/cmd/ColdStorageAddressBookPlugin:1]: It has this option to https://book.getfoundry.sh/reference/cast/cast-send#etherscan-options

File: .env.example [hunk 6]
```diff
@@ -1,16 +1,15 @@
...
 RPC_URL=http://127.0.0.1:8545
 
 
 ########### Wallet info & secrets ###########
 MSCA_FACTORY_OWNER_PRIVATE_KEY=0x9...2
+MSCA_FACTORY_OWNER_ADDRESS=0x0...5
 DEPLOYER_PRIVATE_KEY=0x8...6
 DEPLOYER_ADDRESS=0xB...E
 
-
```
Comment (huaweigu) [inline .env.example:13]: nitpick: I think we intentionally did 2 lines between sections

File: README.md [hunk 7]
```diff
@@ -5,41 +5,49 @@ This repository includes support for both the Hardhat and Foundry frameworks. Go
...
+We are using [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) structure to automatically generate releases.
+
 ## Export Interface
 To export contract bytecode & ABI for programmable wallet:
 1. Execute `forge build src/msca/6900/v0.7 --force --extra-output-files abi evm`. Replace v0.7 with v0.8 if you want v0.8.
 2. Execute `make abigen`
 3. Interface files will appear under `abigen` folder
 4. After changes are merged, release new repository tag
-5. update `buidl-wallet-contracts` go mod version of programmable-wallet, and import bytecode from `abigen` folder
+5. update `smart-wallet-contracts` go mod version of programmable-wallet, and import bytecode from `abigen` folder
```
Comment (huaweigu) [inline README.md:46]: typo, buidl

File: script/000_ContractAddress.sol [hunk 8]
```diff
@@ -0,0 +1,14 @@
+pragma solidity 0.8.24;
```
Comment (huaweigu) [inline script/000_ContractAddress.sol:1]: missing copyright

**Review by huaweigu — APPROVED** [review]

**Comment by huaweigu** [comment]
LGTM. @dlim-circle could you please take another look at your convenience?

**Review by dlim-circle — APPROVED** [review]
@ashutosh-ukey Think you asked previously, did you also wanted to deploy the mainnet contracts for plugins that doesn't need owners?

**Comment by ashutosh-ukey** [comment]
@dlim-circle I will deploy those ones locally and update our internal docs, but I was planning on bunching together all of the mainnet MSCA audited contract deployment logs into one big PR later.

**Pull request merged**